components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -172,11 +172,48 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
                     "Error while retrieving oauth issuer for the app with clientId: " + consumerKey, e);
         }
 
-        synchronized ((consumerKey + ":" + authorizedUserId + ":" + scope + ":" + tokenBindingReference).intern()) {
-            AccessTokenDO existingTokenBean = null;
+        AccessTokenDO existingTokenBean = null;
+        OAuthAppDO oAuthAppDO = (OAuthAppDO) tokReqMsgCtx.getProperty(OAUTH_APP);
+        String tokenType = oauthTokenIssuer.getAccessTokenType();
+
+        /*
+        This segment handles access token requests that neither require generating a new token
+        nor renewing an existing one. Instead, it returns the existing token if it is still valid.
+        As a result, no synchronization lock is needed for these operations.
+        Additionally, this segment should strictly avoid any write or update operations.
+        It was introduced to minimize traffic at the token issuance lock for certain token requests.
+         */
+        if (!(JWT.equalsIgnoreCase(tokenType) && getRenewWithoutRevokingExistingStatus() &&
```

Comment:
Apart from "Default" and "JWT", token type there can be custom token types too, won't there any issue with this logic?

Reply:
Given that only two types of token classes implement OauthTokenIssuer interface in this repo, how come custom token flow exists?

Reply:
Regardless of the token type, when a token can be requested without revoking or consent, this flow can be ustilized to issue the existing token.